src/test/java/com/powsybl/networkconverterserver/NetworkConverterTest.java

```diff
@@ -0,0 +1,84 @@
+package com.powsybl.networkconverterserver;
+
+import com.powsybl.commons.datasource.ReadOnlyDataSource;
+import com.powsybl.iidm.import_.Importers;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.network.store.client.NetworkStoreService;
+import org.apache.commons.compress.utils.IOUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.ArgumentMatchers.any;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(NetworkConverterController.class)
+@ContextConfiguration(classes = {NetworkConverterApplication.class, NetworkConverterService.class})
+public class NetworkConverterTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Mock
+    private RestTemplate caseServerRest;
+
+    @Mock
+    private RestTemplate geoDataRest;
+
+    @Autowired
+    private NetworkConverterService networkConverterService;
+
+    @MockBean
+    private NetworkStoreService networkStoreClient;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void test() throws Exception {
+        ClassLoader classLoader = getClass().getClassLoader();
+        try (InputStream inputStream = classLoader.getResourceAsStream("testCase.xiidm")) {
+            byte[] networkByte = IOUtils.toByteArray(inputStream);
+            networkConverterService.setCaseServerRest(caseServerRest);
+            networkConverterService.setGeoDataRest(geoDataRest);
+            given(caseServerRest.exchange(any(String.class), any(HttpMethod.class), any(HttpEntity.class), any(Class.class)))
+                    .willReturn(new ResponseEntity<>(networkByte, HttpStatus.OK));
+            InputStream byteArrayInputStream = new ByteArrayInputStream(networkByte);
+            Network network = Importers.loadNetwork("test.xiidm", byteArrayInputStream);
```

Comment:
Don't use Importers.loadNetwork. Use Importers.importData and ResourceDataSource instead to avoid reading the XIIDM into an intermediate byte array.